tests/api/test_account.py

```diff
@@ -143,6 +143,30 @@ def test_query_user(staff_api_client, customer_user, permission_manage_users):
     assert address['phone'] == user_address.phone.as_e164
 
 
+def test_non_staff_user_can_only_see_his_user_data(user_api_client,
+                                                   staff_user):
+    query = """
+    query User($id: ID!) {
+        user(id: $id) {
+            email
+        }
+    }
+    """
+    ID = graphene.Node.to_global_id('User', staff_user.id)
+    variables = {'id': ID}
+    response = user_api_client.post_graphql(query, variables)
+    content = get_graphql_content(response)
+    data = content['data']['user']
+    assert data is None
+    user = user_api_client.user
```

Comment:
Please put a comment before each case or split it up between the two tests, so it will be nice and clear what we are testing and where